src/CoreShop/Bundle/StoreBundle/CoreExtension/StoreOptionProvider.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * CoreShop.
+ *
+ * This source file is subject to the GNU General Public License version 3 (GPLv3)
+ * For the full copyright and license information, please view the LICENSE.md and gpl-3.0.txt
+ * files that are distributed with this source code.
+ *
+ * @copyright  Copyright (c) CoreShop GmbH (https://www.coreshop.org)
+ * @license    https://www.coreshop.org/license     GNU General Public License version 3 (GPLv3)
+ */
+
+namespace CoreShop\Bundle\StoreBundle\CoreExtension;
+
+use CoreShop\Bundle\StoreBundle\Doctrine\ORM\StoreRepository;
+use Pimcore\Model\DataObject\ClassDefinition\Data;
+use Pimcore\Model\DataObject\ClassDefinition\DynamicOptionsProvider\SelectOptionsProviderInterface;
+
+class StoreOptionProvider implements SelectOptionsProviderInterface
+{
+    /** @var StoreRepository */
+    private $repository;
+
+    /**
+     * @param StoreRepository $repository
+     */
+    public function __construct(StoreRepository $repository)
```

Comment:
use PHP8 Constructor property promotion

Reply:
Fixed. I do not like this syntax but doesn't matter...